pkg/nodeconfig/nodeconfig.go

```diff
@@ -293,20 +294,58 @@ func (nc *nodeConfig) getWICDServiceAccountSecret() (*core.Secret, error) {
 		return nil, err
 	}
 	// Go through all the secrets in the WMCO namespace, and find the token secret which contains the auth credentials
-	// for the WICD ServiceAccount. This secret's name will always have the form:
-	// ${service_account_name}-token-${random_string}.
-	tokenSecretPrefix := "windows-instance-config-daemon-token-"
+	// for the WICD ServiceAccount
 	var filteredSecrets []core.Secret
 	for _, secret := range secrets.Items {
-		if strings.HasPrefix(secret.Name, tokenSecretPrefix) {
+		if secret.Type != core.SecretTypeServiceAccountToken {
+			// skip non-serviceAccount token secrets
+			continue
+		}
+		if secret.Annotations[core.ServiceAccountNameKey] == windows.WicdServiceName {
 			filteredSecrets = append(filteredSecrets, secret)
 		}
 	}
-	if len(filteredSecrets) != 1 {
-		return nil, fmt.Errorf("expected 1 secret with '%s' prefix, found %d", tokenSecretPrefix, len(filteredSecrets))
+	if len(filteredSecrets) == 1 {
+		return &filteredSecrets[0], nil
+	}
+	if len(filteredSecrets) > 1 {
+		return nil, fmt.Errorf("expected 1 secret for SA '%s', found %d", windows.WicdServiceName,
+			len(filteredSecrets))
 	}
-	return &filteredSecrets[0], nil
+	// no secret token found for WICD service account, create one
+	return nc.createWICDServiceAccountTokenSecret()
+}
 
+// createWICDServiceAccountTokenSecret creates a secret with a long-lived API token for the WICD ServiceAccount and
+// waits for the secret data to be populated
+func (nc *nodeConfig) createWICDServiceAccountTokenSecret() (*core.Secret, error) {
+	ctx := context.TODO()
+	err := nc.client.Create(ctx, secrets.GenerateServiceAccountTokenSecret(nc.wmcoNamespace, windows.WicdServiceName))
+	if err != nil {
+		return nil, fmt.Errorf("error creating secret for WICD ServiceAccount: %w", err)
+	}
+	secret := &core.Secret{}
+	// wait for the secret data to be populated
+	err = wait.PollUntilContextTimeout(ctx, retry.Interval, retry.Timeout, false,
```

Comment:
Please reduce the timeout, 10 minutes is a very long time to be blocking on this.
How long do you expect for this to take? I'd expect this to be almost instant.
Any more than a 30 second to 1 minute wait should cause the reconciliation to fail and requeue.

Reply:
Also please change this to an immediate poll. Not doing so adds in a constant 15 second delay within the reconciliation process that may not be necessary.